metrics/options.go

```diff
@@ -17,10 +18,20 @@ const (
 type Option func(*options) error
 
 type options struct {
-	errorHandler     errors.ErrorHandler
-	dsdEndpoint      string
-	metricSampleRate float64
-	tags             []string
+	errorHandler errors.ErrorHandler
+	dsdEndpoint  string
+	sampleRate   float64
+	tags         []string
+}
+
+// MetricOptions represents a configuration option for metrics.
+type MetricOptions func(*metricOptions) error
+
+// metricOptions tags and sample rate opts.
+type metricOptions struct {
+	errorHandler errors.ErrorHandler
```

Comment:
Is is necessary to be able to define an `errorHandler` for a specific metric? Could it be simpler to rely on the global one?

```suggestion
```